src/lib/swapper/swappers/CowSwapper/endpoints.ts

```diff
@@ -105,9 +114,21 @@ export const cowApi: SwapperApi = {
     // For the slippage actually to be enforced, the final message to be signed needs to have slippage deducted.
     // Failure to do so means orders may take forever to be filled, or never be filled at all.
     const quoteBuyAmount = quote.buyAmount
-    const slippageDeductedBuyAmount = bn(quoteBuyAmount).minus(
-      bn(quoteBuyAmount).times(slippageTolerancePercentageDecimal),
+
+    const hasAffiliateFee = bn(tradeQuote.affiliateBps).gt(0)
+
+    // Remove affiliate fees off the buyAmount to get the amount after affiliate fees, but before slippage bips
+    const buyAmountAfterAffiliateFeesCryptoBaseUnit = hasAffiliateFee
+      ? bn(quoteBuyAmount)
+          .times(bn(1).minus(convertBasisPointsToDecimalPercentage(tradeQuote.affiliateBps)))
+          .toFixed(0)
+      : quoteBuyAmount
+    const buyAmountAfterAffiliateFeesAndSlippageCryptoBaseUnit = bn(
+      buyAmountAfterAffiliateFeesCryptoBaseUnit,
     )
+      .minus(bn(quoteBuyAmount).times(slippageTolerancePercentageDecimal))
+      .toFixed(0)
+
```

Comment:
why aren't we using the same `getValuesFromQuoteResponse` helper here as well? And it looks like this spot is taking into account slippage, which I would think the first quote would want to as well?

Reply:
Per convo, will follow up with helper consolidation and associated unit tests.